### PR TITLE
Fixes #384 add advacned pk parameter vignettes

### LINKED ISSUE #384: Add advanced pk paramaters description to vignette
Extend pk-analysis.Rmd with the information on creation and usage of advanced pk parameters

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
at least there is something
Please add at your convenience

File: vignettes/pk-analysis.Rmd [hunk 1]
```diff
@@ -90,3 +94,85 @@ exportPKAnalysesToCSV(pkAnalyses = pkAnalysis, filePath = csvPKAnalysisPath)
...
+
+The `{ospsuite}` package also supports user-defined PK Parameters, e.g. PK Parameter that can be tailored to specific project needs.
+This feature is useful when calculating PK Parameters for specific time intervals, or to apply PK Parameters for output not using 
+the dimension `Concentration`. 
+
+A User-Defined PK Parameter is always based on an existing PK-Parameter and simply extends the way the output is calcualted.
+It is not possible at the moment to define your own `formula`.
+
+The utility `addUserDefinedPKParameter` creates and returns an instance of the `UserDefinedPKParameter` which can then be parameterized
+for specific requirements.
```
Comment (Yuri05) [inline vignettes/pk-analysis.Rmd:109]: Here we could directly link to the online help:
The utility [`addUserDefinedPKParameter`](https://www.open-systems-pharmacology.org/OSPSuite-R/reference/addUserDefinedPKParameter.html) creates and returns an instance of the [`UserDefinedPKParameter`](https://www.open-systems-pharmacology.org/OSPSuite-R/reference/UserDefinedPKParameter.html) which can then be parameterized for specific requirements.

File: vignettes/pk-analysis.Rmd [hunk 2]
```diff
@@ -14,9 +14,13 @@ knitr::opts_chunk$set(
 )
 ```
 
-## Calculating PK parameters
+## Calculating PK   parameters
```
Comment (Yuri05) [inline vignettes/pk-analysis.Rmd:17]: additional blank on purpose?

File: vignettes/pk-analysis.Rmd [hunk 3]
```diff
@@ -90,3 +94,85 @@ exportPKAnalysesToCSV(pkAnalyses = pkAnalysis, filePath = csvPKAnalysisPath)
...
+* The calculated value of the user defined PK Parameter will be interpreted as value in the base unit of this dimension, e.g. here "kg/dm²"
+* If the display unit differs from the base unit of the dimension, calculated value will be converted into the target display unit
+
+E.g. in the example above:
+
+* First, the C_max value of the quantity will be calculated (because the user defined parameter is derived from C_max)
+* From the display unit `mg/m²`, the dimension will be estimated ("Dose per body surface area")
+* The calculated C_max value will be interpreted as a value in `kg/dm²`
+* Finally, the calculated value will be converted to `mg/m²` and set in the user defined PK Parameter.
+
```
Comment (Yuri05) [inline vignettes/pk-analysis.Rmd:176]: Very good explanation and example!
This is one of the core features with userDef pk params which is not self-explaining at all

File: vignettes/pk-analysis.Rmd [hunk 4]
```diff
@@ -90,3 +94,85 @@ exportPKAnalysesToCSV(pkAnalyses = pkAnalysis, filePath = csvPKAnalysisPath)
...
+# Specifies start application (4th) and end applicaiton (5th)
+ myCMax$startApplicationIndex <- 4
+ myCMax$endApplicationIndex <- 5
+```
+
+#### Update Dimension of a standard parameter for correct visualization
+
+PK parameters currently always assume that the underlying quantity is a concentration and do not check for units.
+If the quantity of interest has another dimension, creating a user-defined PK paramter `AND` set its display unit to any valid unit of the target dimension can do the trick
```
[the remark this answers is not in the document]
Comment (msevestre) [reply vignettes/pk-analysis.Rmd:148]: done

File: vignettes/pk-analysis.Rmd (hunk 2 above)
Comment (msevestre) [reply vignettes/pk-analysis.Rmd:17]: reverted

File: vignettes/pk-analysis.Rmd (hunk 3 above)
Comment (msevestre) [reply vignettes/pk-analysis.Rmd:176]: that's your explanation :)

File: vignettes/pk-analysis.Rmd [hunk 5]
```diff
@@ -90,3 +94,85 @@ exportPKAnalysesToCSV(pkAnalyses = pkAnalysis, filePath = csvPKAnalysisPath)
...
+# Create a new parameter based on the standard C_max parameter
+ myCMax <- addUserDefinedPKParameter(
+   name = "MyCMax",
+   standardPKParameter = StandardPKParameter$C_max
+)
+
+# Specifies start application (4th) and end applicaiton (5th)
+ myCMax$startApplicationIndex <- 4
+ myCMax$endApplicationIndex <- 5
+```
```
[the remark this answers is not in the document]
Comment (msevestre) [reply vignettes/pk-analysis.Rmd:143]: done

File: R/unitConverter.R [hunk 6]
```diff
@@ -83,12 +82,13 @@
         sourceUnit          = yUnit,
         molWeight           = molWeight,
         molWeightUnit       = ospUnits$`Molecular weight`$`g/mol`
-      ))) %>%
```
Comment (msevestre) [inline R/unitConverter.R:86]: @IndrajeetPatil I ran the styler and this was changed => you'll have to rebase the other PR.
Can we add `run the styler` to our CODING STANDARD :)

**Comment by msevestre** [comment]
@Yuri05 Merging after your review so that @IndrajeetPatil can rebase his work. Please create another issue for changes if needed

File: vignettes/pk-analysis.Rmd (hunk 3 above)
Comment (Yuri05) [reply vignettes/pk-analysis.Rmd:198]: really? cool :)

**Pull request merged**